opennms-dao-mock/src/main/java/org/opennms/netmgt/dao/mock/MockMonitoredServiceDao.java

```diff
@@ -188,4 +188,15 @@ public OnmsMonitoredService getPrimaryService(final Integer nodeId, final String
     public List<OnmsMonitoredService> findByServiceTypeAndIpInterfaceId(String serviceTypePrefix, List<Integer> ipInterfaceIds) {
         throw new UnsupportedOperationException("Not yet implemented!");
     }
+
+    @Override
+    public List<OnmsMonitoredService> findSimilarServicesOnInterface(Integer nodeId, InetAddress ipAddress, String svcNamePrefix) {
```

Comment:
Can we please do this by class name of the associated service instead of relying on a service name/prefix.

Main concern with this id, that it would kill the poller pattern feature, which would allow for a neat integration if different config types.

Reply:
> Can we please do this by class name of the associated service instead of relying on a service name/prefix.
> 
> Main concern with this id, that it would kill the poller pattern feature, which would allow for a neat integration if different config types.

I checked how poller pattern feature and that seems to fit what we need here.
Let's chat on this.  Currently poller patterns only work with `pattern` context which is not available on UI.
When I tried  `config-type` with poller pattern, ran into below error, may be pattern doesn't support `-`
```
Error executing command: named capturing group is missing trailing '>' near index 22
^DevicConfig-(?<config-type>.*)$
```
Also need to see how to integrate this with manual poller execution (within DeviceConfigServiceImpl).

We can tackle using poller pattern for multiple services in another issue.

Reply:
We can take care of this in another issue.